src/game/Spell.cpp

```diff
@@ -4264,7 +4264,10 @@ SpellCastResult Spell::CheckCast(bool strict)
             }
 
             if (!dispelTarget)
-                return SPELL_FAILED_NOTHING_TO_DISPEL;
+                if (m_spellInfo->Id == 24406) // Hunter's Improved Mend Pet
```

Comment:
basically you want to revert all of this right here. This is the reason why;
> It currently also ignores the "Nothing to dispel" check/error and just casts a dispel regardless -- this particular behaviour is undesirable

is happening. what you wanna try instead is change line https://github.com/cmangos/mangos-classic/pull/222/files#diff-b4db11cbf2813cd24b3c165885fc08a7R4249 to
if (m_spellInfo->Effect[0] == SPELL_EFFECT_DISPEL && m_spellInfo->Id != 24406) // Ignore Hunter's Improved Mend Pet)

by doing that you also avoid having to execute the whole shabang of dispel check, and you stop returning spell cast ok prematurely. possibly the basepoints or the effectmiscvalues for this effect, particularly for classic, could be used in a way we don't yet understand.

The whole dispel check was just recently implemented, it seems like the classic branch is a bit different to the other branches. Perhaps it could be interesting to see the behaviour of the spell/talent on the other cores, if it wasn't removed in tbc that is.